### DISCUSSION TITLE
custom regions file

### DISCUSSION THREAD

**Discussion by e-coral** [Ruitulyu/KAS-Analyzer#13] [Q&A] [answered]
Hi, 

For some of the analyses I want to run (e.g. diff), the documentation says I will need to use a custom regions file for peaks. However, I can't find anything that says what this file needs to contain. Please can someone explain what this file is, or show an example of a custom regions file? 

(I have already successfully run peakscalling, normalize and heatmap - maybe the output of one of these is relevant?)

**Comment by Ruitulyu** [comment]
Hi,

To perform differential KAS-seq analysis, it's necessary to define the
regions of interest using the '-r' option. If you set '-r' to 'peak', you
must supply a custom region file. This file should include all regions you
wish to analyze for KAS-seq signals.

Best,
Ruitu

On Wed, Nov 29, 2023 at 10:12 AM E Croot ***@***.***> wrote:

> Hi,
>
> For some of the analyses I want to run (e.g. diff), the documentation says
> I will need to use a custom regions file for peaks. However, I find
> anything that says what this file needs to contain. Please can someone
> explain what this file is, or show an example of a custom regions file?
>
> (I have already successfully run peakscalling, normalize and heatmap -
> maybe the output of one of these is relevant?)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Ruitulyu/KAS-Analyzer/discussions/13>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APDPVGEG5TL76OLL5DQRDDLYG5NGDAVCNFSM6AAAAAA77XMNYKVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZVHEYDMMRXGU>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Nutricula, based in Chicago.

**Comment by e-coral** [comment]
Thanks, but I'm still not clear about what this file should include.

I am trying to identify peaks that are bigger/smaller/differentially present in one set of replicates compared to another.

Could I supply the peaks.[sharp/broad].bed files as custom region files for this purpose?

**Reply by Ruitulyu** [reply, marked as answer]
Sure, I think you can supply the peaks.bed files as custom region files this purpose.

**Reply by e-coral** [reply]
Thank you for your help. 

Unfortunately, I have run into another issue with this command. 

Firstly, thank you for the updates so far that have added some t2t compatibility to KAS-Analyzer - they have been very useful!

The script diff_regions.sh is still not t2t compatible, and gives `Error: unsupported assembly id: hs1` 

I have tried to resolve this by adding the code
`&& [[ $assemblyid != "hs1" ]]` and `if [[ $assemblyid == "hs1" ]] ;then genomesize=3117292070 elif...` to the relevant places in the script.

This has prevented the `Error: unsupported assembly id: hs1` problem, however, the diff command now gives a different error (which may or may not be related to my attempt to make this command compatible with t2t):

```Generate the new index of O1_S4_R1_001.ext150.nor.bigWig ...

[E::hts_hopen] Failed to open file O1_S4_R1_001.ext150.nor.bigWig
[E::hts_open_format] Failed to open file "O1_S4_R1_001.ext150.nor.bigWig" : Exec format error
samtools index: failed to open "O1_S4_R1_001.ext150.nor.bigWig": Exec format error
```
The documentation says to use the generated bigWig files for diff, but according to the error, samtools seems unable to handle this filetype - perhaps the error is inaccurate, and indicating a different problem?

If you have any ideas about the cause of this, or suggestions for what I can try to resolve this issue, please let me know.
